### PR TITLE
Update files

### PR DESCRIPTION
**Pull request by samwaseda**
`environment.yml` looked very old so I updated it along with README

### CODE REVIEW

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by jan-janssen** [comment]
@samwaseda Can you take a look at this?
```
File /usr/share/miniconda/envs/test/lib/python3.12/site-packages/ase/cell.py:74, in Cell.new(cls, cell)
     71 if cell is None:
     72     cell = np.zeros((3, 3))
---> 74 cell = np.array(cell, float)
     76 if cell.shape == (3,):
     77     cell = np.diag(cell)

TypeError: float() argument must be a string or a real number, not 'dict'
```

**Comment by samwaseda** [comment]
> @samwaseda Can you take a look at this?

I got what the problem is but I haven't been able to run the notebook in my environment. Give me a moment

**Comment by samwaseda** [comment]
Why didn't the tests run?

File: .github/ci_support/build_notebooks.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ i=0;
 for f in $(find . -name "*.ipynb" | sort -n); do
     cd $(dirname $f);
     notebook=$(basename $f);
-    papermill ${notebook} ${notebook%.*}-out.${notebook##*.} -k "python3" || i=$((i+1));
+    papermill ${notebook} ${notebook%.*}-out.${notebook##*.} || i=$((i+1));
```
Comment (jan-janssen) [inline .github/ci_support/build_notebooks.sh]:
```suggestion
    papermill ${notebook} ${notebook%.*}-out.${notebook##*.} -k "python3" || i=$((i+1));
```

**Comment by samwaseda** [comment]
I think the problem is related to how the stored job is used, but it's probably part of the `pack` and `unpack` problem, so maybe we should solve it first and finish this PR.

**Comment by jan-janssen** [comment]
I more or less did the same changes in https://github.com/pyiron/pyiron-publication-template/pull/23 and https://github.com/pyiron/pyiron-publication-template/pull/25 . The Python 3.12 update does not work as demonstrated in https://github.com/pyiron/pyiron-publication-template/pull/26 so I switched to Python 3.11 .

**Comment by samwaseda** [comment]
But we could keep it open no?

**Comment by jan-janssen** [comment]
> But we could keep it open no?

The issue is fixed, is not it?

**Comment by samwaseda** [comment]
No. I think the other PR is failing for the same reason

**Comment by jan-janssen** [comment]
> No. I think the other PR is failing for the same reason

The other PR is failing with Python 3.12 but 3.11 worked fine and is already merged with `main`.

**Pull request closed without merging**